### PR TITLE
many: remove usages of deprecated io/ioutil package

### PR DESCRIPTION
**Pull request by MiguelPires**
The first commit removes usages of the deprecated io/ioutil package. This is mostly a simple replacement. The exception is `testutil/lowlevel.go` which mocked some of these functions and so had slightly larger changes. Similarly, some of its usages used `FileInfo` in ways that had to be slightly altered because they now use a `DirEntry`.
The second commit prevent new usages of the deprecated methods from being introduced. There's other similar checks in run-checks and I started to refactor them but opted against do that in this PR.

### CODE REVIEW

File: run-checks [hunk 1]
```diff
@@ -207,6 +207,22 @@ if [ "$STATIC" = 1 ]; then
         exit 1
     fi
 
+    echo "Checking for usages of deprecated io/ioutil"
+    got=""
+    for dir in $(go list -f '{{.Dir}}' ./... | grep -v '/vendor/' ); do
+        # shellcheck disable=SC2063
+        s="$(grep -nP io/ioutil "$dir"/*.go || true)"
```
Comment (bboozzoo) [inline run-checks:214]: thanks for this

**Review by bboozzoo — APPROVED** [review]
Thank you

File: cmd/snap/cmd_validate_test.go [hunk 2]
```diff
@@ -21,12 +21,11 @@ package main_test
...
 import (
 	"fmt"
-	"io/ioutil"
+	"io"
 	"net/http"
 
+	main "github.com/snapcore/snapd/cmd/snap"
 	"gopkg.in/check.v1"
-
-	"github.com/snapcore/snapd/cmd/snap"
```
[the remark this answers is not in the document]
Comment (MiguelPires) [reply cmd/snap/cmd_validate_test.go:29]: Hm I think the script only runs `gofmt` which AFAIK sorts imports that are grouped together but doesn't separate 3rd party and local imports. I think `goimports` does what we want here, I can open a new PR to add a few lines to `run-checks`. Thanks for raising this

File: overlord/devicestate/crypto.go [hunk 3]
```diff
@@ -34,9 +33,9 @@ import (
 
 func generateRSAKey(keyLength int) (*rsa.PrivateKey, error) {
 	// The temporary directory is created with mode
-	// 0700 by ioutil.TempDir, see:
-	//   https://github.com/golang/go/blob/master/src/io/ioutil/tempfile.go#L84
-	tempDir, err := ioutil.TempDir(os.TempDir(), "snapd")
+	// 0700 by os.MkdirTemp, see:
+	//   https://github.com/golang/go/blob/master/src/os/tempfile.go#L98
```
[the remark this answers is not in the document]
Comment (MiguelPires) [reply overlord/devicestate/crypto.go:37]: I agree, the previous link was outdated so it's probably best

File: sandbox/seccomp/seccomp.go [hunk 4]
```diff
@@ -58,7 +58,7 @@ func (scp *secCompProbe) actions() []string {
 	return scp.probedActions
 }
 
-var ioutilReadFile = ioutil.ReadFile
+var ioutilReadFile = os.ReadFile
```
Comment (ZeyadYasser) [inline sandbox/seccomp/seccomp.go:61]: tiny nitpick
```suggestion
var osReadFile = os.ReadFile
```

File: snap/container.go
Comment (ZeyadYasser) [inline snap/container.go:1]: This fix was merged https://github.com/snapcore/snapd/pull/13759, maybe a rebase is needed?

**Review by ZeyadYasser — APPROVED** [review]
LGTM, Thank you!

I think it might be a good idea to also remove `io/ioutil` from the ci checks:
https://github.com/snapcore/snapd/blob/1d280d9f6098d6baad7bf5478af7a522a6ac5847/.golangci.yml#L83-L86

File: cmd/snap/cmd_auto_import.go [hunk 5]
```diff
@@ -185,7 +184,7 @@ func autoImportFromAllMounts(cli *client.Client) (int, error) {
 	return added, nil
 }
 
-var ioutilTempDir = ioutil.TempDir
+var ioutilTempDir = os.MkdirTemp
```
[the remark this answers is not in the document]
Comment (MiguelPires) [reply cmd/snap/cmd_auto_import.go:187]: Thank you, fixed this now

**Comment by MiguelPires** [comment]
> LGTM, Thank you!
> 
> I think it might be a good idea to also remove `io/ioutil` from the ci checks:
> 
> https://github.com/snapcore/snapd/blob/1d280d9f6098d6baad7bf5478af7a522a6ac5847/.golangci.yml#L83-L86

Thanks for catching this, I missed it in my grepping. I'll squash the commits and rebase since there are two +1s

**Pull request merged**